src/controllers/keyboard/keyboardcontroller.h

```diff
@@ -0,0 +1,74 @@
+#ifndef KEYBOARDCONTROLLER_H
+#define KEYBOARDCONTROLLER_H
+
+#include "controllers/controller.h"
+#include "controllers/keyboard/keyboardcontrollerpreset.h"
+#include "keyboardeventfilter.h"
+
+// KeyboardController is the controller for the keyboard, holding a reference to the KeyboardEventFilter, which is
+// sent a signal when a new preset --new keyboard mapping-- is loaded.
+class KeyboardController : public Controller {
+    Q_OBJECT
+  public:
+    KeyboardController(KeyboardEventFilter* pKbdEventFilter);
+    virtual ~KeyboardController();
```

Comment:
'override' also applies to virtual destructors. Use 'override' instead of 'virtual' in derived classes. This ensures that the base class actually declares a virtual destructor.